civis/futures.py

```diff
@@ -285,3 +292,386 @@ def cancel(self):
+        If ``True``, a counter will be added to the ``script_name`` to create
+        the script names for each submission.
+
+    See Also
+    --------
+    civis.APIClient.scripts.post_containers
+    """
+    def __init__(self, docker_image_name="civisanalytics/datascience-python",
+                 docker_image_tag="latest",
+                 repo_http_uri=None,
+                 repo_ref=None,
+                 git_credential_id=None,
+                 script_name=None,
+                 required_resources=None,
+                 time_zone=None,
+                 hidden=True,
+                 params=None,
+                 arguments=None,
+                 max_n_retries=0,
+                 client=None,
+                 polling_interval=None,
+                 inc_script_names=False):
+        self.docker_image_name = docker_image_name
+        self.docker_image_tag = docker_image_tag
+        self.repo_http_uri = repo_http_uri
+        self.repo_ref = repo_ref
+        self.git_credential_id = git_credential_id
+        self.required_resources = required_resources
+        self.time_zone = time_zone
+        self.params = params
+        self.arguments = arguments
+
+        if required_resources is None:
+            required_resources = {'cpu': 1024, 'memory': 1024}
+        self.required_resources = required_resources
+
+        if script_name is None:
+            date_str = datetime.datetime.today().strftime("%Y-%m-%d")
+            script_name = ("ContainerPoolShellExecutorScript "
+                           "{}".format(date_str))
+
+        super().__init__(script_name=script_name,
+                         hidden=hidden,
+                         client=client,
+                         max_n_retries=max_n_retries,
+                         polling_interval=polling_interval,
+                         inc_script_names=inc_script_names)
+
+    def _create_job(self, script_name, arguments=None, cmd=None):
+        # Combine instance and input arguments into one dictionary.
+        # Use `None` instead of an empty dictionary.
+        combined_args = (self.arguments or {}).copy()
+        combined_args.update((arguments or {}))
+        if not combined_args:
+            combined_args = None
+
+        # Submit a request to Civis to make the container script object.
+        job = self.client.scripts.post_containers(
+            name=script_name,
+            required_resources=self.required_resources,
+            repo_http_uri=self.repo_http_uri,
+            repo_ref=self.repo_ref,
+            docker_command=cmd,
+            docker_image_name=self.docker_image_name,
+            docker_image_tag=self.docker_image_tag,
+            time_zone=self.time_zone,
+            hidden=self.hidden,
+            params=self.params,
+            arguments=combined_args,
+            git_credential_id=self.git_credential_id,
+        )
+
+        return job
+
+
+class CustomPoolExecutor(_CivisExecutor):
```

Comment:
How about `CustomScriptExecutor`? It's shorter, and there's not really a "pool".